sympy/core/numbers.py

```diff
@@ -418,7 +418,23 @@ def __rdivmod__(self, other):
         return divmod(other, self)
 
     def __round__(self, *args):
-        return round(float(self), *args)
+        if len(args)>1:
+            raise TypeError("__round__ takes at most 2 arguments (%s given)" % str(len(args)+1))
+        elif len(args)==0:
+            arg = 0
+        else:
+            arg = args[0]
+        if arg < 0:
+            pre = str(round(self,arg))
+        else:
+            decimal.getcontext().prec = len(str(self))
+            upto = decimal.Decimal(1)/(decimal.Decimal(10)**(arg))
+            pre = decimal.Decimal(str(self)).quantize(upto, decimal.ROUND_HALF_UP)
```

Comment:
It would be better to use mpmath rather than decimal. decimal can have different rounding semantics than floats, due to they way they are implemented (see for instance the example at https://github.com/sympy/sympy/issues/12038).